### ISSUE TITLE
new Int8Array(n).toArray throws TypeError

### ISSUE THREAD

**Issue by weihsiu** [scala-js/scala-js#1759]
```
new Int8Array(10).toArray
```
throws TypeError when testing using phantom.js
```
caused by: TypeError: Int8ArrayConstructor is not a constructor (evaluating 'this.u.constructor(this.u)')
```

in a browser, it throws
```
Uncaught TypeError: Constructor Int8Array requires 'new'
```

it looks like in scalajsenv.js, "this.u.constructor" is a constructor for Int8Array and is not called with "new".

**Comment by gzm0** [comment]
I don't know if this is webkit not properly supporting the full typedarray specification or us using it incorrectly. In any case it needs investigation, therefore assigning to 0.6.4 for the time being.

**Comment by sjrd** [comment]
It seems the browser is right to throw a `TypeError`:
http://www.ecma-international.org/ecma-262/6.0/#sec-%typedarray%-intrinsic-object-%typedarray%

> If NewTarget is undefined, throw a TypeError exception.

We should add a `new`. Now would that fix the issue in PhantomJS? I don't know.

**Comment by weihsiu** [comment]
i don't think the fix is complete. in scalajsenv.js, "new" was added to the ECMAScript6 branch, not the prototype branch. i am still getting the same errors in both phantom.js and chrome after using scalajs 0.6.4.

**Comment by sjrd** [comment]
Fixed in #1867.
